buildSrc/subprojects/plugins/src/main/kotlin/gradlebuild/publish-public-libraries.gradle.kts

```diff
@@ -54,15 +54,15 @@ artifacts {
             "pom",
             null,
             Date(),
-            generatePom.pomFile,
+            generatePom.get().pomFile,
             generatePom))
 }
 
-fun Project.configureUploadArchivesTask(generatePom: GeneratePom) {
-    tasks.getByName<Upload>("uploadArchives") {
+fun Project.configureUploadArchivesTask(generatePom: TaskProvider<GeneratePom>) {
+    tasks.withType<Upload>().named("uploadArchives").configure {
```

Comment:
This won't work due to the type returned by the Kotlin `withType<>` extension method.

Reply:
This also doesn't work because the lazy APIs ignore tasks that are created by rules.

Reply:
I would put this back and leave it eager.  I raised https://github.com/gradle/gradle-native/issues/718

`Upload` is getting replaced by the other publish plugins, so we may want to refactor our build to use `maven-publish` and we could avoid this.